### PR TITLE
[FIX] web: can't edit subscription with end date beyond 200 years in …

### PR DESCRIPTION
**Pull request by Florimond**
…the future

-Import a subscription with end date beyond 200 in the future (ex: 2500-01-01).
-Open the subscription and click the Edit button.

Before this commit:

A stacktrace appears indicating that a date is not valid. It's not possible to
edit the subscription.

After this commit:

The invalid date is replaced with the max date, allowing the user to edit the
subscription.

OPW: 2079696

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Florimond** [comment]
@nim-odoo

**Comment by sswapnesh** [comment]
The longest branch name I have ever seen :wink: 

`odoo-dev:12.0-opw-2079696-fix-cant-edit-subscriptions-with-end-date-beyond-200-years-in-the-future-fhu`

(no offence, Just kidding :smiley: )

File: addons/web/static/src/js/widgets/date_picker.js [hunk 1]
```diff
@@ -25,12 +25,11 @@ var DateWidget = Widget.extend({
     init: function (parent, options) {
         this._super.apply(this, arguments);
 
-        this.name = parent.name;
```
Comment (nim-odoo) [inline addons/web/static/src/js/widgets/date_picker.js]: beuh...

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 2]
```diff
@@ -2718,6 +2718,47 @@ QUnit.module('basic_fields', {
...
+            data: this.data,
+            arch:'<form><field name="foo"/><field name="date"/></form>',
+            translateParameters: {  // Avoid issues due to localization formats
+                date_format: '%m/%d/%Y',
+            },
+            res_id: 1,
+            viewOptions: {
+                mode: 'edit',
+            },
+            debug: true,
```
Comment (nim-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: remove debug

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 3]
```diff
@@ -2718,6 +2718,47 @@ QUnit.module('basic_fields', {
...
+        assert.strictEqual($('.bootstrap-datetimepicker-widget:visible').length, 1,
+            "datepicker should be opened");
+
+        // focus another field
+        form.$('.o_field_widget[name=foo]').click().focus();
+
+        assert.strictEqual($('.bootstrap-datetimepicker-widget:visible').length, 0,
+            "datepicker should close itself when the user clicks outside");
+
+        //form.destroy();
```
Comment (nim-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: add this back

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 4]
```diff
@@ -2718,6 +2718,47 @@ QUnit.module('basic_fields', {
         form.destroy();
     });
 
+    QUnit.test('date field: toggle datepicker far in the future', function (assert) {
+        assert.expect(3);
+
+        this.data.partner.records = [{
+            id: 1,
+            date: "6017-02-03",
```
Comment (nim-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: set the date until '9999-12-31'

File: addons/web/static/src/js/widgets/date_picker.js (hunk 1 above)
Comment (nim-odoo) [reply addons/web/static/src/js/widgets/date_picker.js]: Do not remove this part

**Pull request closed without merging**